includes/model/UserWorkLog_model.php

```diff
@@ -121,6 +121,41 @@ function UserWorkLog_create($userWorkLog)
     return $result;
 }
 
+/**
+ * @param array|int $shift
+ */
+function UserWorkLog_from_shift($shift)
+{
+    $shift = is_array($shift) ? $shift : Shift($shift);
+
+    if ($shift['start'] > time()) {
+        return;
+    }
+
+    $room = Room($shift['RID']);
+    foreach ($shift['ShiftEntry'] as $entry) {
+        if ($entry['freeloaded']) {
+            continue;
+        }
+        $type = AngelType($entry['TID']);
+
+        UserWorkLog_create([
+            'user_id'        => $entry['UID'],
+            'work_timestamp' => $shift['start'],
+            'work_hours'     => ($shift['end'] - $shift['start']) / 60 / 60,
```

Comment:
This should be doubled on night shifts using the configured rules.

Reply:
Hmm, fair point but this would mean the worklog hours wont change even if the admin changes the night shifts times but as we have no data about it but that's better than not adding them ;)